### PR TITLE
Improve UX of deploy's remember credentials

### LINKED ISSUE #2860: Improve UX of "Remember credentials" toggle in deployment
__What should we do?__

Improve the UX of "Remember credentials" toggle in the deployment overlay for Cloud.

- [ ] Make 'Remember credentials' toggle always available (currently not available for self-managed target with no OATH)
- [ ] Save credentials onClose, regardless if the deployment succeed 

__Why should we do it?__

This is part of a bigger effort to improve the UX of the deployment journey (see #2707)

----

Child of #2707

### PR DESCRIPTION
**Pull request by smbea**
Closes #2860

See example:
1. Try wrong credentials (can't deploy) -> close overlay -> credentials saved
2. Try right credentials with invalid diagram -> deploy fails but credentials are saved

https://user-images.githubusercontent.com/25825387/161419793-eb5e9966-ec36-4032-a180-da72d5081017.mov

### CODE REVIEW

**Comment by andreasgeier** [comment]
@smbea Can you please provide a screen recording? That would save me the hassle of setting up SaaS and SM for testing.

**Comment by MaxTru** [comment]
In #2860 we speced to:

>Save credentials onClose, regardless if the deployment succeed

However, this now only saves credentials when I try to deploy (success or failure). Am I missing something?

**Comment by smbea** [comment]
> In #2860 we spec'd to:
> 
> > Save credentials onClose, regardless if the deployment succeed
> 
> However, this now only saves credentials when I try to deploy (success or failure). Am I missing something?

It should save on closing too. If you open the overlay, type something and click out of it, it should save it. Does this not happen?

**Comment by MaxTru** [comment]
> > In #2860 we spec'd to:
> > > Save credentials onClose, regardless if the deployment succeed
> > 
> > 
> > However, this now only saves credentials when I try to deploy (success or failure). Am I missing something?
> 
> It should save on closing too. If you open the overlay, type something and click out of it, it should save it. Does this not happen?

No...

doesntSave

**Comment by MaxTru** [comment]
The Validation to ensure that only valid C8 SaaS URLs are put in (when having C8 SaaS selected) does not work anymore on this branch. See
Screenshot_20220404_092419

**Comment by smbea** [comment]
@MaxTru can you try this again?  I tested it with an artifact and it worked, so it should work for you 😅

**Comment by MaxTru** [comment]
With the recent changes - it works now :+1:

File: client/src/plugins/zeebe-plugin/deployment-plugin/__tests__/DeploymentPluginSpec.js [hunk 1]
```diff
@@ -263,6 +263,20 @@ describe('<DeploymentPlugin> (Zeebe)', () => {
   });
 
 
+  it('should save tab on close', async () => {
```
Comment (MaxTru) [inline client/src/plugins/zeebe-plugin/deployment-plugin/__tests__/DeploymentPluginSpec.js:266]: The testcase is named `on close` but in fact it does `deploy`? Is that right?

Comment (MaxTru) [reply client/src/plugins/zeebe-plugin/deployment-plugin/__tests__/DeploymentPluginSpec.js:266]: :+1:

**Comment by MaxTru** [comment]
@smbea and I walked through this. @smbea will investigate whether we can improve test coverage of the adjusted features

**Comment by smbea** [comment]
> @smbea and I walked through this. @smbea will investigate whether we can improve test coverage of the adjusted features

@MaxTru i was able to add one more test case for the connection check on input change. And it already has a test case for saving the credentials on closing

**Review by MaxTru — APPROVED** [review]
:+1: Good improvement

**Pull request merged**